### ISSUE TITLE
This episodic memory didn't consider a case when the steps of a episode is under trajectory_legnth

### ISSUE THREAD

**Issue by verystrongjoe** [fshamshirdar/pytorch-rdpg#2]
This episodic memory didn't consider a case when the steps of a episode is under trajectory_length so that it could add trajectory of uncorrelated two episodes data into memory. If I am wrong, let me know!! 

    def append(self, state0, action, reward, terminal, training=True):
        self.trajectory.append(Experience(state0=state0, action=action, reward=reward, state1=None, terminal1=terminal)) # 
        if len(self.trajectory) >= self.max_episode_length:
            self.memory.append(self.trajectory)
            self.trajectory = []

**Comment by fshamshirdar** [comment]
In this implementation trajectory length could be in any size, does not matter whether it reaches the trajectory_length or not (I brought the code from keras-rl as mentioned at the beginning of the file). As far as I know, having different trajectory_lengths would give us faster training and have a better generalization. I will actually consider implementing this feature in future.

**Comment by verystrongjoe** [comment]
Yes, it could.trajectory length can vary in each episode. I mean that your code can allow steps of two or more episodes can be included in same trajectory. right?

**Comment by fshamshirdar** [comment]
I didn't quite understand your question, but if you mean that the code lets us have different trajectory lengths, yes it does. an episode can have any trajectory length below the max_trajectory_length.
